azure-functions/acbs-function/mappings/facility/facility-guarantee-amend.test.js

```diff
@@ -0,0 +1,46 @@
+const facilityGuaranteeAmend = require('./facility-guarantee-amend');
+
+describe('facilityGuaranteeAmend', () => {
+  describe('when there is an amendment amount', () => {
+    const amendment = { amount: 123.456 };
+
+    it('returns the target amount rounded to 2 decimal places', () => {
+      const result = facilityGuaranteeAmend(amendment);
+      expect(result).toEqual({ guaranteedLimit: 123.46 });
+    });
+  });
+
+  describe('when there is a guarantee expiry date', () => {
+    const amendment = { facilityGuaranteeDates: { guaranteeExpiryDate: '2022-01-01' } };
+    it('returns the guarantee expiry date', () => {
+      const result = facilityGuaranteeAmend(amendment);
+      expect(result).toEqual({ expirationDate: '2022-01-01' });
+    });
+  });
+
+  describe('when there is an amendment amount and a guarantee expiry date', () => {
+    const amendment = { amount: 123.456, facilityGuaranteeDates: { guaranteeExpiryDate: '2022-01-01' } };
+    it('returns the target amount rounded to 2 decimal places and the guarantee expiry date', () => {
+      const result = facilityGuaranteeAmend(amendment);
+      expect(result).toEqual({ guaranteedLimit: 123.46, expirationDate: '2022-01-01' });
+    });
+  });
+
+  describe('when there is no amendment amount or guarantee expiry date', () => {
+    const amendment = {};
+    const result = facilityGuaranteeAmend(amendment);
+    expect(result).toEqual({});
+  });
+
+  describe('when there are extra fields', () => {
```

Comment:
Can you please add an assertion for following:

* Exceptional handling
* When `facilityGuaranteeDates` has an empty object.
* When the amount is in wrong data type.

Reply:
We check exception handling -- existing pattern is to catch and return an empty object (I've added an `it` to the test which was missing!)

I've added a check when `facilityGuaranteeDates` is an empty object

Regarding `amount` is the wrong datatype -- this is currently not handled at all well (one of the reasons I looked at adding zod!). Technically, as this is a unit test, we should be mocking `to2Decimals` -- I'll update in line with this

Reply:
Understood regarding above points thanks for that.
Wondering if we can remove this `to2DecimalsResponse` and rely on the actual response?